src/main/java/org/killbill/billing/plugin/api/PluginProperties.java

```diff
@@ -91,6 +92,11 @@ public boolean apply(final PluginProperty input) {
         return Strings.isNullOrEmpty(pluginPropertyString) ? null : pluginPropertyString;
     }
 
+    public static String findAndDecodePluginPropertyValue(final String pluginPropertyName, @Nullable final Iterable<PluginProperty> properties) {
+        String undecodedValue = findPluginPropertyValue(pluginPropertyName, properties);
+        return Strings.isNullOrEmpty(undecodedValue) ? null : UTF8UrlDecoder.decode(undecodedValue).toString();
```

Comment:
`undecodedValue` can never be empty, right (since `findPluginPropertyValue` already checks)?